File: Aman_Bhaiya_DSA_List/Arrays/LQ28.java
package Aman_Bhaiya_DSA_List.Arrays;

public class LQ28{
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length -1;
        int mxw = (r-l) * Math.min(height[l],height[r]);
        while(l < r){
            mxw = Math.max(mxw,((r-l) * Math.min(height[l],height[r])));
            if(height[l]<= height[r])
                l++;
            else
                r--;
        }
        
        return mxw;
    }
}

File: Aman_Bhaiya_DSA_List/Arrays/LQ28.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        mxw = (r-l) * min(height[l],height[r])

        while l < r:
            mxw = max(mxw,((r-l) * min(height[l],height[r])))
            if height[l] <= height[r]:
                l += 1
            else:
                r -= 1
        
        return mxw